types/defines/pipeline-transform.d.ts

```diff
@@ -11,3 +11,18 @@ export abstract class PipelineTransform {
    */
   public transformJson(data: object[]): Promise<object[]>;
 }
+
+export type PipelineResponse = {
```

Comment:
Sorry for commenting on API design here given that the code is probably already written, but what is this return type communicating that wouldn't be equally-well communicated by a rejected promise / thrown Error?

In what cases will `success` be false? In what cases will `error` be filled in vs the `Promise` being rejected?

I'd have expected `send` to just return a `Promise<void>` unless there was something more than success/error to be returned.

Reply:
For example that's what the Queues `send()` method returns: https://github.com/cloudflare/workerd/blob/main/src/workerd/api/queue.h#L68